x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/cache/full/SearchableSnapshotsPrewarmingIntegTests.java

```diff
@@ -265,7 +267,7 @@ public void onFailure(Exception e) {
                 }
             });
         }
-
+        maxUploadTasksCreated.await();
```

Comment:
Suggest using `safeAwait` to bound the wait and handle interruptions properly.

```suggestion
        safeAwait(maxUploadTasksCreated);
```